### PR TITLE
Use the selected texture filtering in AnimatedSprite2D frame preview

### LINKED ISSUE #57550: AnimatedSprite2D animation frame previews are displayed without regard to the selected filter
### Godot version

v4.0.alpha.custom_build (bf12719cc)

### System information

Windows 11, Geforce 1660

### Issue description

In 4.x previews are blurry, although the sprite is displayed correctly on the scene.

https://user-images.githubusercontent.com/10379487/152166892-b03eb757-7c3d-4ac1-b13f-ee4cc60fe72a.mp4

In 3.x previews are crisp:

image

### Steps to reproduce

-open bat scene;
-select AnimatedSprite2D node.

### Minimal reproduction project

[PreviewTest_G4.zip](https://github.com/godotengine/godot/files/7987096/PreviewTest_G4.zip)

*Bugsquad edit (keywords for easier searching): SpriteFrames, nearest, nearest-neighbor*

### PR DESCRIPTION
**Pull request by hakro**
Fixes #57550

### CODE REVIEW

**Comment by akien-mga** [comment]
See #57596 and the discussion there, I'm not sure this fix handles everything that should be (inheritance of filtering mode, AnimatedSprite3D).

**Comment by hakro** [comment]
Oops ! Didn't see that other PR. My bad.

Indeed, this one only solves the AnimatedSprite2D issue, and automatically takes the value of texture_filter, and passes it to the SpriteFrames preview when it updates.

It doesn't make too much sense with AnimatedSprite3D, since texture_filter, is a CanvasItem property, which AS3D does not inherit from.

https://user-images.githubusercontent.com/3624853/221428755-ed2b216b-11be-4f89-a775-b40098f3f0b9.mp4

But I guess we can close this PR if it's not the right way to go, and the issue is being worked already.

Cheers !

File: editor/plugins/sprite_frames_editor_plugin.cpp [hunk 1]
```diff
@@ -1103,6 +1103,12 @@ void SpriteFramesEditor::_update_library(bool p_skip_selector) {
 		} else {
 			autoplay->set_pressed(String(edited_anim) == autoplay_name);
 		}
+
+		// Use same image filtering in frame previews as the sprite
+		AnimatedSprite2D *animated_sprite_2d = Object::cast_to<AnimatedSprite2D>(animated_sprite);
+		if (animated_sprite_2d) {
+			frame_list->set_texture_filter(animated_sprite_2d->get_texture_filter());
```
Comment (dalexeev) [inline editor/plugins/sprite_frames_editor_plugin.cpp:1110]:
```suggestion
			frame_list->set_texture_filter(animated_sprite_2d->get_texture_filter());
			split_sheet_preview->set_texture_filter(animated_sprite_2d->get_texture_filter());
```

**Comment by dalexeev** [comment]
Note that the SpriteFrames editor can be used for a single `SpriteFrames` resource, without `AnimatedSprite{2D,3D}` node.

File: editor/plugins/sprite_frames_editor_plugin.cpp (hunk 1 above)
Comment (hakro) [reply editor/plugins/sprite_frames_editor_plugin.cpp]: Thank you.
I've just added that.

**Comment by hakro** [comment]
I just pushed a few changes, and it mostly works, but there are a couple of edge cases which are yet to be fixed.
So now, the `SpriteFrames` resource gets it's filtering from the CanvasItem, if it's attached to one.

If it's a stand alone `SpriteFrame`, it gets its filtering option from the project setting.
Unfortunately the mapping done with the `switch` clause is not very elegant, but it's necessary. That's because `CanvasItem::TextureFilter` enums do not match the filter enum in the project settings. So a mapping is needed.

There are still 2 problems though.
- On a stand alone `SpriteFrame`, only new animations inherit what's set in the project settings. Existing ones do not update. Need to find a way to update those.
- For some reason, when the CanvasItem's filter is set to Inherit, the `SpriteFrames` does not get the same filter, while it works for all the other values.

**Comment by dalexeev** [comment]
I'm not sure if `_update_library()` is the right place to do this.

**Comment by hakro** [comment]
> I'm not sure if `_update_library()` is the right place to do this.

What would you suggest ?

**Comment by dalexeev** [comment]
I would suggest `SpriteFramesEditorPlugin::edit()` as this is where the edited resource is changed. Also note that in the case of `AnimatedSprite3D`, some default filter must also be applied, as in the case of standalone `SpriteFrames` (because `Node3D` does not have a texture filter).

But personally, I think it's overcomplicated. I don't see any problem with `TEXTURE_FILTER_NEAREST_WITH_MIPMAPS` (#74341). See also #67426. There, initially the author suggested a separate setting, but in the end we decided that this was not necessary.

In the `TileMap`/`TileSet` editor, this is also not configurable and does not depend on the project settings:

https://github.com/godotengine/godot/blob/61d2c855114c824f5ca27ded0a1fa71cc7b21134/editor/plugins/tiles/tile_map_editor.cpp#L2216

https://github.com/godotengine/godot/blob/61d2c855114c824f5ca27ded0a1fa71cc7b21134/editor/plugins/tiles/tile_set_editor.cpp#L781

@Calinou What do you think about it?

**Comment by dalexeev** [comment]
#57550 (duplicate of #74337 or vice versa) was fixed by #74341. This should be closed as a simpler solution has been merged. Nevertheless, thanks for your contribution!

**Comment by akien-mga** [comment]
I still think this ought to work out of the box and there's a deeper bug that will need to be looked into: https://github.com/godotengine/godot/pull/57596#issuecomment-1429835376

But for the time being #74341 is likely "good enough".

**Pull request closed without merging**